### PR TITLE
Add time to email pane and change attachment list order

### PR DESCRIPTION
**Pull request by flywire**

### CODE REVIEW

**Comment by flywire** [comment]
Note: 01729f3 should be used in preference to ba389f5 since suggested first and change is identical. Other parts of PRs are different.

**Comment by flywire** [comment]
If you doubt this simplifies the build process it then try it on a clean PC.

**Comment by Dijji** [comment]
Agree with attachment list change: good spot. Agree with idea of adding time, but code could be more simply expressed as:
`((DateTime)Date).ToString("g")`
which produces the same 'short date + space + short time' result.

The readme changes, I don't really understand.
Why `Open a local folder` when double-clicking on the solution file opens Visual Studio and the project?
Why press `Ctrl+Shift+B`? It doesn't do anything on my machine, the hotkey is F6.
Why switch to Release mode at all? If you do, why use the Configuration Manager, which you haven't told them how to get to, and not just change the drop down on the toolbar from Debug to Release?

**Comment by Dijji** [comment]
I forgot to say, thank you for your first contribution as a pull request! All the better for being mostly right 👍

**Comment by flywire** [comment]
Thank you, we've both after the same thing. This is the first time I've used either VS or C# and it has caused a bit of grief bringing up the environment, esp loading GBs of rubbish, My previous experience is with procedural languages and VB but nothing oop. Using VS to clone repository seems to give the cleanest download and put it in a nice structure.

I'd welcome any direction and I'm sure there is a better path but I wonder if the latest VS release is different (?).

2019-12-07 23_51_01-XstReader - Microsoft Visual Studio

Microsoft Visual Studio V16_4_0

btw - I'm finding the code comments hard to follow. I was expecting something more VB like eg [Your First C# Windows Form](https://www.homeandlearn.co.uk/csharp/csharp_s1p5.html) with a Form.cs object.

**Comment by Dijji** [comment]
It turns out the same version of Visual Studio 2019 as you, it is just that I am running the Community Edition, and I have a different hotkey on the same menu.

If we cannot rely on consistent hotkeys, we had better say Build/Build Solution, and Debug/Start Debugging instead.

It is my code comment that was obscure. I was merely observing that
`((DateTime)Date).ToShortDateString() + " " + ((DateTime)Date).ToShortTimeString()`
and
`((DateTime)Date).ToString("g")`
yield the same result, but the latter is the crisper expression.

I would have expected there to be detailed click throughs of using Visual Studio and C# out there somewhere: some people seem to love creating such things.

**Comment by Dijji** [comment]
By the way, what I would like to do is to take some of your changes and not others. However, it is very difficult to partially apply a pull request. By git convention, each group of files making up a change would be made in a different branch, and each would be proposed as a separate pull request.

In this case, what I currently propose to do is to make changes in my own code base: reordering attachments, as proposed, adding time, as proposed, but modified as I mentioned, and build instructions, change only hotkeys to menu directions.

Then ultimately, I will reject your pull request as superseded.

**Comment by flywire** [comment]
* As I said I suspect the hotkeys change is very recent (days), the click-through tutes I found like the one I linked had the same hotkeys you used
* Agree your code is clean
* Sorry for messy PR, no worries about rejecting it - main thing is it demonstrated issues and I learn code better with a meaningful purpose
  * Can you close the PR/PR commits with a comment in your new commit?
  * Do I need a separate branch for each PR? (As you can see I made separate commits.)
* btw, Pointless having a fork with no changes so I'll delete my fork after your commit is made
* Still trying to work out where the header is above the contents pane so I can tweak it lol

**Comment by Dijji** [comment]
Closing as superseded by a revised version of the changes checked into master

**Comment by Dijji** [comment]
Yes, a pull request for each commit that you want merged into master is usual. It is difficult to separate multiple commits within one pull request, and there is no intended way to separate changes made within one commit.

**Comment by flywire** [comment]
You missed it:
> Do I need a **separate branch** for each PR?

**Comment by Dijji** [comment]
I think the answer is basically yes. You can avoid it, but I wouldn't bother. Here are the instructions that I found.

You actually CAN do this without creating another branch, but it takes a bit of playing around.
Here's the steps:

1. Identify the two commit ranges you want to pull. Here's what i'll use for an example:
(other/master) A -> B -> C -> D -> E (yours/master)
Let's say that you want to pull B and C in one request, and D & E in another.
1. Make a pull request. Have the left side ("Base") be commit A. For the right side ("head"), type in the commit number of C.
2. Write the description for your first request.
3. Make another request. For the base, type in the commit number of C, and for the head, put E (yours/master).
4. Write the description.

As I see it, the pull request sees commit C as a branch point. Or something.
